src/components/settings/members/CreateInviteDialog.tsx

```diff
@@ -0,0 +1,193 @@
+import { forwardRef, useState } from 'react'
+import { gql } from '@apollo/client'
+import { useFormik } from 'formik'
+import { object, string } from 'yup'
+import styled from 'styled-components'
+
+import { theme } from '~/styles'
+import { Button, Dialog, DialogRef, Typography } from '~/components/designSystem'
+import { TextInputField } from '~/components/form'
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+import { CreateInviteInput, Lago_Api_Error, useCreateInviteMutation } from '~/generated/graphql'
+import ErrorImage from '~/public/images/maneki/error.svg'
+import { addToast, LagoGQLError, useCurrentUserInfosVar } from '~/core/apolloClient'
+import { GenericPlaceholder } from '~/components/GenericPlaceholder'
+
+gql`
+  mutation createInvite($input: CreateInviteInput!) {
+    createInvite(input: $input) {
+      id
+      token
+    }
+  }
+`
+
+export interface CreateInviteDialogRef extends DialogRef {}
+
+export const CreateInviteDialog = forwardRef<DialogRef>((_, ref) => {
+  const { translate } = useInternationalization()
+  const { currentOrganization } = useCurrentUserInfosVar()
+  const [inviteToken, setInviteToken] = useState<string>()
+  const [createInvite, { error }] = useCreateInviteMutation({
+    onCompleted(res) {
+      if (res?.createInvite?.token) {
+        setInviteToken(res.createInvite.token)
+      }
+    },
+  })
+
+  const formikProps = useFormik<CreateInviteInput>({
+    initialValues: {
+      email: '',
+    },
+    validationSchema: object().shape({
+      email: string().email().required(''),
+    }),
+    enableReinitialize: true,
+    validateOnMount: true,
+    onSubmit: async (values, formikBag) => {
+      const result = await createInvite({
+        variables: {
+          input: {
+            ...values,
+          },
+        },
+        refetchQueries: ['getInvites'],
+      })
+
+      const { errors } = result
+
+      const apiError = !errors ? undefined : (errors[0]?.extensions as LagoGQLError['extensions'])
+
+      if (
+        !!apiError &&
+        apiError?.code === Lago_Api_Error.UnprocessableEntity &&
+        !!apiError?.details?.invite
+      ) {
+        formikBag.setFieldError('email', translate('text_63208c701ce25db781407456'))
+      }
+    },
+  })
+
+  return (
+    <Dialog
+      ref={ref}
+      title={translate('text_63208c701ce25db78140748f')}
+      description={
+        inviteToken
+          ? translate('text_63208c701ce25db78140743a', {
+              organizationName: currentOrganization?.name,
+            })
+          : translate('text_63208c701ce25db78140749b')
+      }
+      onOpen={() => {
+        formikProps.validateForm()
+      }}
+      onClickAway={() => {
+        setInviteToken('')
+        formikProps.resetForm()
+        formikProps.validateForm()
```

Comment:
Why do you need that ?

Reply:
yeah it's a bit terrible. I got the same issue than last time where the form is not reset and validation is wrong even if I had the good settings in formik.

What I could do it to move all those "resets" in the onOpen hook then it's in only one place at least